include/triSYCL/vec/detail/vec.hpp

```diff
@@ -146,6 +157,15 @@ class alignas(alignment_v<::trisycl::vec<DataType, NumElements>>)
         std::make_index_sequence<std::tuple_size<decltype(xTuple)>::value>());
   }
 
+
+  template<typename T, typename T2, typename F>
+  static auto apply_ternary_functor(const T x, const T2 y, const T2 z, F f) {
+    auto xTuple = flatten_to_tuple<T>(x);
+    auto yTuple = flatten_to_tuple<T>(y);
+    return apply_binary_functor_impl(xTuple, yTuple, f,
```

Comment:
That part is suspicious : there is a z which is never used and `apply_binary_functor_impl` is called inside the `apply_ternary_functor` function.

Reply:
Indeed... Good catch!
Probably some leftover of WIP action.
I removed it.